nibabel/cifti2/tests/test_cifti2.py

```diff
@@ -273,59 +316,77 @@ def test_matrixindicesmap():
     volume2 = ci.Cifti2Volume()
     parcel = ci.Cifti2Parcel()
 
-    assert_is_none(mim.volume)
+    assert mim.volume is None
     mim.append(volume)
     mim.append(parcel)
 
 
-    assert_equal(mim.volume, volume)
-    assert_raises(ci.Cifti2HeaderError, mim.insert, 0, volume)
-    assert_raises(ci.Cifti2HeaderError, mim.__setitem__, 1, volume)
+    assert mim.volume == volume
+    with pytest.raises(ci.Cifti2HeaderError):
+        mim.insert(0, volume)
+    
+    with pytest.raises(ci.Cifti2HeaderError):
+        mim[1] = volume
 
     mim[0] = volume2
-    assert_equal(mim.volume, volume2)
+    assert mim.volume == volume2
 
     del mim.volume
-    assert_is_none(mim.volume)
-    assert_raises(ValueError, delattr, mim, 'volume')
+    assert mim.volume is None
+    with pytest.raises(ValueError):
+        delattr(mim, 'volume')
 
     mim.volume = volume
-    assert_equal(mim.volume, volume)
+    assert mim.volume == volume
     mim.volume = volume2
-    assert_equal(mim.volume, volume2)
+    assert mim.volume == volume2
 
-    assert_raises(ValueError, setattr, mim, 'volume', parcel)
+    with pytest.raises(ValueError):
+        setattr(mim, 'volume', parcel)
 
 
 def test_matrix():
     m = ci.Cifti2Matrix()
-    assert_raises(TypeError, m, setattr, 'metadata', ci.Cifti2Parcel())
-    assert_raises(TypeError, m.__setitem__, 0, ci.Cifti2Parcel())
-    assert_raises(TypeError, m.insert, 0, ci.Cifti2Parcel())
+
+    with pytest.raises(TypeError):
+        m(setattr, 'metadata', ci.Cifti2Parcel())
```

Comment:
This was @demianw's test, added in c2dbd1637fa095f045e21fea224b8b29e913cb71, so I hope he can verify the intent, but my guess is this was supposed to be equivalent to:

```suggestion
        m.metadata = ci.Cifti2Parcel()
```

Reply:
This throws me :
```python
E  ValueError: Not a valid Cifti2MetaData instance.

nibabel/cifti2/cifti2.py:95: ValueError
```

Reply:
Ah, then I'd update it to a `ValueError`. `m(setattr, ...)` really doesn't make any sense.